## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 420855`

Logs: https://github.com/cameronraysmith/nixpkgs-review-gha/actions/runs/15948561968


---
### `x86_64-linux` (sandbox = true)
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>star</li>
  </ul>
</details>


---
### `x86_64-darwin` (sandbox = true)
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>star</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-darwin`</summary>
<details>
<summary>star</summary>
<pre>./IncludeDefine.h:24:10: fatal error: 'omp.h' file not found
   24 | #include <omp.h>
      |          ^~~~~~~
1 error generated.
In file included from streamFuns.cpp:1:
In file included from ./streamFuns.h:4:
In file included from ./Parameters.h:4:
./IncludeDefine.h:24:10: fatal error: 'omp.h' file not found
   24 | #include <omp.h>
      |          ^~~~~~~
1 error generated.
In file included from twoPassRunPass1.cpp:1:
In file included from ./twoPassRunPass1.h:4:
In file included from ./Parameters.h:4:
./IncludeDefine.h:24:10: fatal error: 'omp.h' file not found
   24 | #include <omp.h>
      |          ^~~~~~~
1 error generated.
Makefile:138: Depend.list: No such file or directory
make: *** [Makefile:137: Depend.list] Error 1</pre>
</details>
</details>

--- a/pkgs/by-name/st/star/package.nix
+++ b/pkgs/by-name/st/star/package.nix
@@ -4,6 +4,10 @@
   fetchFromGitHub,
   xxd,
   zlib,
+  llvmPackages,
+  star,
+  testers,
+  nix-update-script,
 }:
 
 stdenv.mkDerivation rec {
@@ -20,19 +24,25 @@ stdenv.mkDerivation rec {
   sourceRoot = "${src.name}/source";
 
   postPatch = ''
-    substituteInPlace Makefile --replace "/bin/rm" "rm"
+    substituteInPlace Makefile --replace-fail "-std=c++11" "-std=c++14"
   '';
 
   nativeBuildInputs = [ xxd ];
 
-  buildInputs = [ zlib ];
+  buildInputs = [ zlib ] ++ lib.optionals stdenv.isDarwin [ llvmPackages.openmp ];
+
+  enableParallelBuilding = true;
+
+  makeFlags = lib.optionals stdenv.hostPlatform.isAarch64 [ "CXXFLAGS_SIMD=" ];
+
+  preBuild = lib.optionalString stdenv.isDarwin ''
+    export CXXFLAGS="$CXXFLAGS -DSHM_NORESERVE=0"
+  '';
 
   buildFlags = [
     "STAR"
     "STARlong"
   ];
-
-  enableParallelBuilding = true;
 
   installPhase = ''
     runHook preInstall
@@ -40,11 +50,22 @@ stdenv.mkDerivation rec {
     runHook postInstall
   '';
 
+  passthru.tests.version = testers.testVersion {
+    package = star;
+    command = "STAR --version";
+  };
+
+  passthru.updateScript = nix-update-script { };
+
   meta = with lib; {
     description = "Spliced Transcripts Alignment to a Reference";
+    longDescription = ''
+      STAR (Spliced Transcripts Alignment to a Reference) is a fast RNA-seq
+      read mapper, with support for splice-junction and fusion read detection.
+    '';
     homepage = "https://github.com/alexdobin/STAR";
     license = licenses.gpl3Plus;
-    platforms = [ "x86_64-linux" ];
+    platforms = platforms.unix;
     maintainers = [ maintainers.arcadio ];
   };
 }

